src/include/distributed/multi_physical_planner.h

```diff
@@ -240,6 +240,14 @@ typedef struct Task
 	char *queryStringLazy;
 	List *perPlacementQueryStrings;
 
+	/*
+	 * queryStringList contains query strings. They should be
+	 * run sequentially. The concatenated version of this list
+	 * will already be set for queryStringLazy, this can be useful
+	 * when we want to access each query string.
+	 */
+	List *queryStringList;
```

Comment:
Ops, yet another piece of information about the query in the task?

Reply:
The motivation is to be able to get each query from the `queryStringLazy`. As there is no easy way to get it back once you concatenate the strings, storing them in a list is one option. @onderkalaci, do you think of another solution?

Reply:
I think it's ok for this to be slightly messy as long as we write accessors for the query string.

Direct access to queryStringLazy / queryForLocalExecution / queryStringList should not be allowed, even for debugging purposes, since that makes refactor more painful.

Reply:
Should this be in copyfuncs? outfuncs?

Reply:
Yeah probably.

Unrelated: perPlacementQueryStrings are not in copy or out funcs. It should also be there I guess.

Reply:
One idea: Can we introduce a new struct, like `TaskQuery`, and we have explicitly have the type of it. Later, we only access through the data via helper functions.
```C
enum TaskQueryType
{
    TASK_QUERY_TEXT,
	TASK_QUERY_LOCAL,
	TASK_QUERY_PER_PLACEMENT,
     TASK_QUERY_LIST
};

typedef struct TaskQuery
{
     TaskQueryType queryType;

     union
     {
         Query *queryForLocalExecution;

	    char *queryStringLazy;
    	List *perPlacementQueryStrings;

	    List *queryStringList;
     }
}

Reply:
queryStringList is independent of the others. it will be set whenever we have a concatenated query string. At the same time queryStringLazy could also be set. With an enum, it seems like it should be only one of them.

Reply:
> queryStringList is independent of the others. it will be set whenever we have a concatenated query string. 

I  think that is our current interpretation in this PR. But, it sounds like it all could (and should?) be distinct. I don't see much value for both `queryStringLazy ` and `queryStringList` set at the same time. We should be able to detect and treat each differently, is there anything that I miss here (e.g., that's not possible or such)?

Anyway, even if that's the case, we can use bitmasks, and can hold multiple types within a single instance.

> But I agree that the query string related fields should be simplified. I would suggest to do it in another PR, what do you think Onder?

If this PR unblocks you on something else, yes that works. But, if you are not planning to improve this immediately after, please open an issue to keep track of.

Thanks!

Reply:
When running a query, it is possible that we will run `TaskQueryString` more than once. If we were to only set `queryStringList` and not set `queryStringLazy` then every time we call `TaskQueryString` we would need to perform `StringJoin` operation. That is the reason we set them both, so that we do not perform `StringJoin` multiple times when `TaskQueryString` is called multiple times.

This PR is necessary for #3636. We need each query string in local execution so that we can execute them one by one. Yet, in the executor we have the concatenated version and it is not easy to split that. 

I can try to refactor query related things in Task to be more understandable in another PR soon, but if this PR doesn't make sense to you I dont want to merge it.

Reply:
> If we were to only set queryStringList and not set queryStringLazy then every time we call TaskQueryString we would need to perform StringJoin operation. That is the reason we set them both, so that we do not perform StringJoin multiple times when TaskQueryString is called multiple times.


Comparing the complexity to store query string(s) in multiple places over re-concat every time, I'd definitely prefer `re-concat every time`, as the relevant code-path is not performance critical.

> I can try to refactor query related things in Task to be more understandable in another PR soon, but if this PR doesn't make sense to you I dont want to merge it.

Well, it seems to complicate things further -- it was already complex, and as a side note I might have contributed this complexity in the past. That's probably why I'm aiming to fix this asap :) 

I think Marco probably thinks that it is becoming more complicated as he referred `this to be slightly messy`.

I'm OK to merge this now as long as we have an issue to track this. A better outcome would be to improve this during this release cycle.

Reply:
@onderkalaci https://github.com/citusdata/citus/issues/3655.